packages/hackmode/lib/index.js

```diff
@@ -185,87 +226,292 @@ class N3hHackMode extends AsyncClass {
+              method: 'handleFetchEntry',
+              dnaAddress: opt.data.dnaAddress,
+              _id: this._createRequestWithBucket(bucketId),
+              requesterAgentId: '',
+              address: entryAddress
+            }
+            this._ipc.send('json', fetchEntry)
+          }
+          return
+
+        case 'handleGetPublishingMetaListResult':
+          // Note: opt.data is MetaListData
+          // Mark my request as resolved and get bucketId from request
+          bucketId = this._checkRequest(opt.data._id)
+          if (bucketId === '') {
+            return
+          }
+
+          // get already known publishing list
+          let knownPublishingMetaList = {}
+          if (bucketId in this._publishedMetaBook) {
+            knownPublishingMetaList = this._publishedMetaBook[bucketId]
+          }
+
+          // Update my book-keeping on what this agent has.
+          // and do a getEntry for every new entry
+          for (let i = 0; i < opt.data.metaList.length; i++) {
+            let metaId = opt.data.metaList[i]
+            if (metaId in knownPublishingMetaList) {
+              continue
+            }
+            let fetchMeta = {
+              method: 'handleFetchMeta',
+              dnaAddress: opt.data.dnaAddress,
+              _id: this._createRequestWithBucket(bucketId),
+              requesterAgentId: '',
+              entryAddress: metaId[0],
+              attribute: metaId[1]
+            }
+            this._ipc.send('json', fetchMeta)
+          }
+          return
+
+        case 'handleGetHoldingMetaListResult':
+          // Note: opt.data is MetaListData
+          // Mark my request as resolved and get bucketId from request
+          bucketId = this._checkRequest(opt.data._id)
+          if (bucketId === '') {
+            return
+          }
+          // get already known publishing list
+          let knownHoldingMetaList = {}
+          if (bucketId in this._storedMetaBook) {
+            knownHoldingMetaList = this._storedMetaBook[bucketId]
+          }
+          // Update my book-keeping on what this agent has.
+          // and do a getEntry for every new entry
+          // for (let entryAddress in opt.data.metaList) {
+          for (let i = 0; i < opt.data.metaList.length; i++) {
+            let metaPair = opt.data.metaList[i]
+            let metaId = this._CatEntryAttribute(metaPair[0], metaPair[1])
+            if (metaId in knownHoldingMetaList) {
+              continue
+            }
+            let fetchMeta = {
+              method: 'handleFetchMeta',
+              dnaAddress: opt.data.dnaAddress,
+              _id: this._createRequestWithBucket(bucketId),
+              requesterAgentId: '',
+              entryAddress: metaPair[0],
+              attribute: metaPair[1]
+            }
+            this._ipc.send('json', fetchMeta)
+          }
           return
       }
     }
 
     throw new Error('unexpected input ' + JSON.stringify(opt))
   }
 
+  /*
```

Comment:
start with two stars for inline [JSDoc](http://usejsdoc.org/)

```javascript
/**
```

Reply:
done